lib/puppet/data_providers/hiera_config.rb

```diff
@@ -99,25 +103,35 @@ def create_configured_data_providers(lookup_invocation, parent_data_provider)
       data_providers = {}
       @config['hierarchy'].each do |he|
         name = he['name']
-        raise Puppet::DataBinding::LookupError, "#{path}: Name '#{name}' defined more than once" if data_providers.include?(name)
+        raise Puppet::DataBinding::LookupError, "#{@config_path}: Name '#{name}' defined more than once" if data_providers.include?(name)
         original_paths = he['paths']
         if original_paths.nil?
           single_path = he['path']
           single_path = name if single_path.nil?
           original_paths = [single_path]
         end
         paths = original_paths.map { |path| interpolate(path, lookup_invocation, false)}
-        provider_name = he['backend']
-        provider_factory = injector.lookup(nil, service_type, PATH_BASED_DATA_PROVIDER_FACTORIES_KEY)[provider_name]
-        raise Puppet::DataBinding::LookupError, "#{@config_path}: No data provider is registered for backend '#{provider_name}' " unless provider_factory
-
         datadir = @config_root + (he['datadir'] || default_datadir)
-        data_providers[name] = create_data_provider(parent_data_provider, provider_factory, name,
-          provider_factory.resolve_paths(datadir, original_paths, paths, lookup_invocation))
+        provider_name = he['backend']
+        data_providers[name] = case provider_name
+        when 'json', 'yaml'
+          Puppet::Pops::Lookup::DataHashFunctionProvider.new(name, parent_data_provider, "#{provider_name}_data", {},
+            resolve_paths(datadir, original_paths, lookup_invocation, ".#{provider_name}"))
+        else
+          # TODO: Remove support for injected provider factories
```

Comment:
No, not really. The comment concerns the specific part of it that is being used here.